### DISCUSSION TITLE
MQTT set commands

### DISCUSSION THREAD

**Discussion by jaspersonneveldt** [rvdbreemen/OTGW-firmware#28] [Q&A] [answered]
Hi,

How can I configure OTGW settings using MQTT (especially room setpoint)? Retrieving values is working fine but how do I set them? If I publish something to, for instance, OTGW/TrSet nothing happens.
I am using home assistant, MQTT auto discovery works fine but it seems also here only sensors are added and nothing to actually control the thermostat or am I missing something?

Thanks!

**Comment by jaspersonneveldt** [comment, marked as answer]
Update: I checked debugging output using telnet and noticed an error about serial port when trying to send a command. See screenshot:

![Screenshot from 2021-02-23 18-49-06](https://user-images.githubusercontent.com/70471326/108886046-cd3bdb00-7608-11eb-82d3-972a4ef6c077.png)

**Reply by rvdbreemen** [reply]
Look like you found a bug. I’ll look at it.

**Reply by rvdbreemen** [reply]
@jaspersonneveldt thanks for reporting the issue. I found the bug and killed it. Just released update as binary.

**Reply by jaspersonneveldt** [reply]
Thanks! I will test it today.

**Reply by jaspersonneveldt** [reply]
Seems to have fixed it, thanks. About my initial question, is this the only way to set the temperature via MQTT? I mean using OTGW/command with value TT=xx or TC=xx.
I guess this would make the use of a thermostat within HA a bit difficult.
![Screenshot from 2021-02-24 09-50-06](https://user-images.githubusercontent.com/70471326/108976165-89d48180-7687-11eb-896a-84dd99a9a116.png)

**Reply by jaspersonneveldt** [reply]
Nevermind, I did not know you can use value templates in HA for the MQTT thermostat parameters. Working fine now using OTGW/command with payload TT=xx
